File: weixiang-ui/src/enums/status.enum.ts
// 状态枚举类型定义
interface StatusEnum {
    code: number;
    description: string;
}

// 状态枚举值
const StatusEnums: { [key: string]: StatusEnum } = {
    NORMAL: { code: 1, description: '正常' },
    DISABLE: { code: 0, description: '禁用' }
};


 export {
    StatusEnums,
 };

File: common/src/main/java/com/hk/enums/StatusEnum.java
package com.hk.enums;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 状态枚举
 *
 * @author huangkun
 */
@Getter
@AllArgsConstructor
public enum StatusEnum {

    NORMAL(1, "正常"),
    DISABLE(0, "禁用"),
    ;
    private final int code;
    private final String description;

    /**
     * 根据code获取
     *
     * @param code
     * @return
     */
    public static StatusEnum getByCode(int code) {
        for (StatusEnum value : StatusEnum.values()) {
            if (value.code == code) {
                return value;
            }
        }
        return null;
    }
}
